auto-ls.zsh

```diff
@@ -30,6 +30,6 @@ auto-ls () {
 zle -N auto-ls
 zle -N accept-line auto-ls
 
-if [[ ${chpwd_functions[(i)auto-ls]} -gt ${#chpwd_functions} ]]; then
+if [[ $chpwd_functions == "" || ${chpwd_functions[(i)auto-ls]} -gt ${#chpwd_functions} ]]; then
```

Comment:
You could use instead:

```
if [[ -z ${#chpwd_functions} || ${chpwd_functions[(I)auto-ls]} -gt 0 ]]
```

The `(I)` flag does return `0` if the key is not found, otherwise it returns the index (base 1).

Reply:
Even shorter:

```
if [[ ${chpwd_functions[(I)auto-ls]} -eq 0 ]]

```

Reply:
Ahh. Didn't know about the (I) option.